pkg/forwarder/tcp.go

```diff
@@ -249,3 +254,60 @@ func (f *tcp) rerouteConn(ctx context.Context, conn net.Conn, clientSession tunn
 	})
 	return nil
 }
+
+// forwardHTTPConn handles HTTP-aware connection forwarding with header/path filtering.
+func (f *tcp) forwardHTTPConn(ctx context.Context, clientConn net.Conn, intercept *manager.InterceptInfo, target netip.AddrPort, wtIntercepts []*manager.InterceptInfo) error {
+	// Create a temporary HTTP interceptor to handle this connection
+	httpInterceptor := &httpInterceptor{
+		interceptor: interceptor{
+			tag:    f.tag,
+			target: target,
+			tCtx:   ctx,
+		},
+		originalTarget: target,
+	}
+
+	// Configure the HTTP interceptor with stream provider and intercept info
+	httpInterceptor.SetStreamProvider(f.streamProvider)
+	httpInterceptor.SetIntercepting(ctx, intercept)
+
+	// Add wiretaps if any
+	for _, wt := range wtIntercepts {
+		httpInterceptor.AddWiretap(wt)
+	}
+
+	// Handle the connection using HTTP logic
+	return httpInterceptor.handleHTTPConn(clientConn)
+}
+
+// DispatchByMechanism implements mechanism-specific per-connection dispatch for TCP.
+// It currently routes HTTP-aware intercepts when requested.
+func (f *tcp) DispatchByMechanism(ctx context.Context, clientConn net.Conn, intercept *manager.InterceptInfo) (bool, error) {
+	var spec *manager.InterceptSpec
+	if intercept != nil {
+		spec = intercept.Spec
+	}
+
+	switch {
+	case spec != nil && spec.HttpMechanism:
```

Comment:
Suggest that you add a `UsesHTTPMechanism(spec *manager.InterceptSpec) bool` similar to the one on Command in the client.